packages/contracts-bedrock/test/Predeploys.t.sol

```diff
@@ -15,17 +46,28 @@ contract PredeploysTest is CommonTest {
             address addr = address(prefix | uint160(i));
             bytes memory code = addr.code;
             assertTrue(code.length > 0);
+
+            bool proxied = _notProxied(addr) == false;
+            bool isPredeploy = _isPredeploy(addr);
+
             // Skip the accounts that do not have a proxy
-            if (
-                addr == Predeploys.LEGACY_ERC20_ETH || addr == Predeploys.GOVERNANCE_TOKEN || addr == Predeploys.WETH9
-                    || addr == Predeploys.MultiCall3 || addr == Predeploys.Create2Deployer || addr == Predeploys.Safe_v130
-                    || addr == Predeploys.SafeL2_v130 || addr == Predeploys.MultiSendCallOnly_v130
-                    || addr == Predeploys.SafeSingletonFactory || addr == Predeploys.DeterministicDeploymentProxy
-                    || addr == Predeploys.MultiSend_v130 || addr == Predeploys.Permit2 || addr == Predeploys.SenderCreator
-                    || addr == Predeploys.EntryPoint
-            ) {
+            if (proxied == false) {
                 continue;
             }
+
+            // Only the defined predeploys have their implementation slot set
+            if (proxied && isPredeploy) {
+                assertEq(
+                    EIP1967Helper.getImplementation(addr),
+                    _predeployToCodeNamespace(addr),
+                    string.concat("Implementation mismatch for ", vm.toString(addr))
+                );
+            }
```

Comment:
It seems like this foregoes the check if the implementation is standard or not. If that is acceptable, then this looks to be comprehensive over migrated checks.

Reply:
Explicitly referring to this comment block
```
// Check that an implementation is set. If the implementation has been upgraded,
// it will be considered non-standard. Ensure that there is code set at the implementation.
```

Reply:
Good thoroughness. This test is not meant to be ran against live networks but only the genesis state. The `check-l2` script was able to run against live networks a long time ago but then it became a CI check against a fresh genesis state. This test is specifically for assertions against genesis state

Reply:
Got it, that makes sense. The check catches any issues either way 😄